src/main/java/com.lelek.springBoot/ApplicationConfiguration.java

```diff
@@ -0,0 +1,37 @@
+package com.lelek.springBoot;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.env.Environment;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+
+import java.util.Objects;
+import java.util.Properties;
+
+@Slf4j
+@Configuration
+@PropertySource("classpath:mail.properties")
```

Comment:
consider moving mail properties in `application.properties`. then you will get them loaded for free. try to rely on standard behaviour that framework provides you